src/components/BuildStartButton/BuildStartButton.tsx

```diff
@@ -0,0 +1,358 @@
+const alignmentPreferences: Array<IParamOption> = [
+  {
+    title: 'Persistent',
+    value: 'PREFER_PERSISTENT',
+    description: 'Prefers latest persistent build version',
+  },
+  {
+    title: 'Temporary',
+    value: 'PREFER_TEMPORARY',
+    description: 'Prefers latest temporary build version',
+  },
+];
+
+const REBUILD_MODE_DEFAULT_INDEX = 1;
+const rebuildModes: Array<IParamOption> = [
+  {
+    title: 'Explicit',
+    value: 'EXPLICIT_DEPENDENCY_CHECK',
+  },
+  {
+    title: 'Implicit',
+    value: 'IMPLICIT_DEPENDENCY_CHECK',
+  },
+  {
+    title: 'Force',
+    value: 'FORCE',
+  },
+];
+
+const persistentPopoverText = 'Standard build, which can be used for product release';
+const temporaryPopoverText =
+  'Test build, which cannot be used for product release and which will be garbage collected during automatic cleaning';
+const keepPodOnFailureAPopoverText =
+  "The builder container won't be destroyed if the build fails and you can use SSH to debug it";
+const buildDependenciesPopoverText = 'Build also dependencies of this build configuration if they are not built yet';
+const descriptionTextPopoverText =
+  'Not implicit (automatically captured) ones - those are used only to determine if the rebuild is required';
+const alignmentPreferencePopoverText = (
+  <>
+    <p>Select temporary build dependency alignment preferences:</p>
+    <br />
+    <div>
+      <b>Persistent</b>
+      <List>
+        <ListItem>Prefers latest persistent build version</ListItem>
+      </List>
+    </div>
+    <br />
+    <div>
+      <b>Temporary</b>
+      <List>
+        <ListItem>Prefers latest temporary build version</ListItem>
+      </List>
+    </div>
+  </>
+);
+const rebuildModePopoverText = (
+  <>
+    <p>Rebuild a previously successful build config when:</p>
+    <br />
+    <div>
+      <b>Explicit</b>
+      <List>
+        <ListItem>The build config has been modified since the last successful build</ListItem>
+        <ListItem>
+          A build config, explicitly defined as a dependency of this one, has been modified since the last successful build
+        </ListItem>
+        <ListItem>
+          There exists a newer, successful build of a build config explicitly defined as a dependency of this one
+        </ListItem>
+      </List>
+    </div>
+    <br />
+    <div>
+      <b>Implicit</b>
+      <List>
+        <ListItem>
+          <b>Explicit</b> criteria plus:
+        </ListItem>
+        <ListItem>
+          There exists a newer version of an implicit dependency (automatically captured from sources such as Indy, MRRC or Maven
+          Central)
+        </ListItem>
+      </List>
+    </div>
+    <br />
+    <div>
+      <b>Force</b>
+      <List>
+        <ListItem>Always</ListItem>
+      </List>
+    </div>
+  </>
+);
+
+export const BuildStartButton = ({ buildConfig, groupConfig, size = 'md' }: IBuildStartButtonProp) => {
+  const [isTempBuild, setIsTempBuild] = useState(false);
```

Comment:
add missing types if possible, for example: `useState<boolean>(false)` (the same for all other cases)